specs/recording-requests.md

```diff
@@ -0,0 +1,21 @@
+# Recording requests
+
+Recording requests allows Developer Proxy to capture requests and related messages that were intercepted by the proxy while in record mode. The captured requests are passed for further processing to the Developer Proxy plugins. This feature is foundational for scenarios such as permission analysis, building tailored SDKs, audit reports, etc.
```

Comment:
Recording is such an overloaded term, I got this idea confused with the idea of recording requests and responses to generate a mock responses file.

Perhaps we should consider being more explicit in the title and opening that we're recording the logging data?

Reply:
This would be totally possible with a dedicated recording plugin! 😀 

Would capturing be clearer?

Reply:
Perhaps we call this Record request logging, something like:
```suggestion
Record request logging allows Developer Proxy to capture request logging activities that occurred while the proxy was in record mode. The captured request logging items are passed for further processing to the Developer Proxy plugins. This feature is foundational for scenarios such as permission analysis, building tailored SDKs, audit reports, etc.
```